src/analyze/analyze-security.c

```diff
@@ -2076,30 +2174,212 @@ static int acquire_security_info(sd_bus *bus, const char *name, struct security_
 }
 
 static int analyze_security_one(sd_bus *bus, const char *name, Table *overview_table, AnalyzeSecurityFlags flags) {
-        _cleanup_(security_info_free) struct security_info info = {
-                .default_dependencies = true,
-                .capability_bounding_set = UINT64_MAX,
-                .restrict_namespaces = UINT64_MAX,
-                ._umask = 0002,
-        };
+        _cleanup_(security_info_freep) SecurityInfo *info = new0(SecurityInfo, 1);
+        if (!info)
+                return log_oom();
+
+        info->default_dependencies = true;
+        info->capability_bounding_set = UINT64_MAX;
+        info->restrict_namespaces = UINT64_MAX;
+        info->_umask = 0002;
```

Comment:
if we initialize a couple of fields right after allocating an object we prefer doing that with C99 structured initialization, i.e. change new0() → new(), and then:

```c
 _cleanup_(security_info_freep) SecurityInfo *info = new(SecurityInfo, 1);

*info = (SecurityInfo) {
        .default_dependencies = true,
        .capability_bounding_set = UINT64_MAX,
        …
};

```

That usually generates better code, since the compiler will set the unlisted fields to zero on its own then.

Reply:
it might make sense to introduce a helper call that allocates these objects and use that everywhere:

```c
SecurityInfo *security_info_new(void) {
        SecurityInfo *info = new(SecurityInfo, 1);

        *info = (SecurityInfo) {
                .default_dependencies = true,
                .capability_bounding_set = UINT64_MAX,
                …
        };

        return info;
}
```

That way code that needs an object will get it properly initialized to defaults and that in a pretty optimized way